vault-config.tf

```diff
@@ -28,21 +30,182 @@ resource "vault_token_auth_backend_role" "nomad-cluster" {
+  type         = "internal"
+  common_name  = "global.nomad"
+  ttl          = "87600h"
+  key_type     = "rsa"
+  key_bits     = 4096
+  organization = "HashiCorp"
+}
+
+resource "vault_mount" "pki_int" {
+  depends_on  = [vault_mount.pki]
+  path                  = "pki_int"
+  type                  = "pki"
+  max_lease_ttl_seconds = 157680000
+}
+resource "vault_pki_secret_backend_intermediate_cert_request" "nomad_int" {
+  depends_on  = [vault_mount.pki]
+  backend     = vault_mount.pki_int.path
+  type        = "internal"
+  key_type     = "rsa"
+  key_bits     = 4096
+  common_name = "global.nomad Intermediate Authority"
+}
+
+resource "vault_pki_secret_backend_root_sign_intermediate" "nomad_root" {
+  backend     = vault_mount.pki.path
+  csr         = vault_pki_secret_backend_intermediate_cert_request.nomad_int.csr
+  format      = "pem"
+  common_name = "global.nomad Intermediate Authority"
+  ttl         = "43800h"
+
+}
+
+resource "vault_pki_secret_backend_intermediate_set_signed" "nomad_int" {
+  backend     = vault_mount.pki_int.path
+  certificate = vault_pki_secret_backend_root_sign_intermediate.nomad_root.certificate
+}
+
+resource "vault_pki_secret_backend_role" "role" {
+  backend          = vault_mount.pki_int.path
+  name             = "nomad-cluster"
+  max_ttl          = 86400
+  allow_ip_sans    = true
+  key_type         = "rsa"
+  key_bits         = 4096
+  allowed_domains  = ["global.nomad", ".nomad", "${data.aws_region.current.name}.nomad"]
+  allow_subdomains = true
+  require_cn       = false
+  generate_lease   = true
+}
+
+resource "vault_policy" "nomad-tls-policy" {
+  name   = "nomad-tls-policy"
+  policy = <<EOF
+path "pki_int/issue/nomad-cluster" {
+  capabilities = ["update"]
+}
+EOF
   depends_on = [
     hcp_vault_cluster.xx_2048_vault
   ]
 }
 
-output "nomad_server_vault_token" {
-  value     = vault_token.nomad_server.client_token
-  sensitive = true
-  depends_on = [
-    hcp_vault_cluster.xx_2048_vault
+# AWS auth for clients and agents
+resource "aws_iam_user" "vault_auth" {
+  name = "aws-iamuser-for-vault-authmethod"
+}
+
+resource "aws_iam_user_policy" "vault_auth" {
+  name   = "aws-iampolicy-for-vault-authmethod"
+  user   = aws_iam_user.vault_auth.name
+  policy = <<EOP
+{
+"Version": "2012-10-17",
+"Statement": [
+ {
+   "Effect": "Allow",
+   "Action": [
+     "ec2:DescribeInstances",
+     "iam:GetInstanceProfile",
+     "iam:GetUser",
+     "iam:ListRoles",
+     "iam:GetRole"
+   ],
+   "Resource": "*"
+ }
+ ]
+}
+EOP
+}
+
+resource "aws_iam_access_key" "vault_mount_user" {
+  user = aws_iam_user.vault_auth.name
+}
+
+resource "vault_auth_backend" "aws" {
+  type = "aws"
+}
+resource "vault_aws_auth_backend_client" "aws_client" {
+  backend    = vault_auth_backend.aws.path
+  access_key = aws_iam_access_key.vault_mount_user.id
+  secret_key = aws_iam_access_key.vault_mount_user.secret
+}
+
+# resource "vault_generic_endpoint" "rotate_initial_aws_credentials" {
```

Comment:
Is this meant to be left commented out?

Reply:
ah.. sorry.. it was to rotate the initial AWS credentials, once the AWS auth backend is configured, but I didn't get to try it. I can remove it if you want, or it can stay as an example.

Reply:
@xiaolin-ninja please let me know how you want to proceed.